go.mod: bump Pebble to caecab07d786

nicktrav opened the pull request:
```
caecab07 db: export some types needed for using lazy values
4510a8c5 db: set sequence numbers in datadriven DB define commands
f07c5e5f db: use mergingIterLevel pointers as heap items
```

Release note: None.

Epic: None.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/94096)
<!-- Reviewable:end -->

rickystewart commented on DEPS.bzl:
```diff
@@ -8990,6 +8990,16 @@ def go_deps():
             "https://storage.googleapis.com/cockroach-godeps/gomod/go.etcd.io/etcd/client/v2/io_etcd_go_etcd_client_v2-v2.305.0.zip",
         ],
     )
+    go_repository(
```
Looks like we now depend on both `go.etcd.io/etcd/raft/v3` as well as `go.etcd.io/raft/v3`. I don't know if this is a huge deal, but getting rid of the extra dependency would be good.

nicktrav replied:
Could this be some fallout from #93190? I assume the intention of that PR was to remove the dependency on the former? cc: @pavelkalinnikov

Let me see if I can resolve this issue separately.

rickystewart reviewed (approved):
Keep an eye on Bazel CI *as well as* GitHub CI for this one please, I'm not sure if the new dependency will cause something to fail. You can pull me in with any questions.

rickystewart commented:
Love to not have to deal with the `vendor` submodule any more 👍

sumeerbhola reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/94096)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @pavelkalinnikov and @rickystewart)

___
*[`DEPS.bzl` line 8993 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/94096#-NJqEiAWZxbx3dKq-QoV-r1-8993:-NJqFpRP532YtGP2OGVk:b-wyj35z) ([raw file](https://github.com/cockroachdb/cockroach/blob/d3191c772e0cac0f1c917d399d2cd9f93debe453/DEPS.bzl#L8993)):*
<details><summary><i>Previously, nicktrav (Nick Travers) wrote…</i></summary><blockquote>

Could this be some fallout from #93190? I assume the intention of that PR was to remove the dependency on the former? cc: @pavelkalinnikov

Let me see if I can resolve this issue separately.
</blockquote></details>

I don't see any imports for the former on CockroachDB master (based on `git grep`), but my goland was quite insistently adding the import on a recent PR (so I had to close and reopen in emacs to undo). Is this a fallout of https://github.com/cockroachdb/cockroach/pull/93190?


<!-- Sent from Reviewable.io -->

sumeerbhola reviewed (approved):
@nicktrav thanks for this bump!
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> module the raft dependency issue

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/94096#-:-NJqGl5M-jPcM_CgPNRP:byl7ija)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @pavelkalinnikov and @rickystewart)



<!-- Sent from Reviewable.io -->

rickystewart commented:
My understanding is #93190 removed the dependency on the old package name `go.etcd.io/etcd/raft/v3` and replaced it everywhere with the new name `go.etcd.io/raft/v3`. I don't know where the import would have been resurrected from since `pebble`'s `go.mod` [doesn't seem to include it?](https://github.com/cockroachdb/pebble/blob/master/go.mod)

rickystewart commented:
According to CI you need to `go mod tidy`.

nicktrav commented:
> According to CI you need to go mod tidy.

Comes back clean for me when run locally :/ If I `go mod tidy` on `master`, it comes back with a diff. Same thing for me on both of my macOS and Linux machines.

I would not expect for _this PR_ to touch anything related to Raft. It should only be bumping the Pebble SHA. I ran `./scripts/bump-pebble.sh`.

nicktrav commented:
I think I see what happened. There were generated proto files lying around that had the old import paths, which `go mod tidy` was seeing and pulling in the old dep. Running `./dev generate` fixed this.

I shall amend this commit.

nicktrav committed:
go.mod: bump Pebble to caecab07d786

```
caecab07 db: export some types needed for using lazy values
4510a8c5 db: set sequence numbers in datadriven DB define commands
f07c5e5f db: use mergingIterLevel pointers as heap items
```

Release note: None.

nicktrav commented:
The one test failure looks like #93958. Going to go ahead and land.

TFTRs!

bors r=rickystewart,sumeerbhola
